app/renderer/components/Inspector/PageObjectPackageParser.js

```diff
@@ -27,100 +26,108 @@ export default class PageObjectPackageParser {
-          }
-        }
-      });
-    }
+              // Populate the list of arguments of the method like (a, b, c)
+              let arglist = '()';
+              if (childPO.methods[childMethodName].args) {
+                method.returnType.methods[childMethodName].args = childPO.methods[childMethodName].args;
+                arglist = '(' + childPO.methods[childMethodName].args.map((a) => a.name).join(',') + ')';
+              }
 
-    // For root Page Object
-    const rootMarkerNode = findNodeAtLocation(rootNode, ['root']);
-    if (rootMarkerNode !== undefined && rootMarkerNode.value === true) {
-      const rootNode = new PageObjectTreeNode(pageObjectName, rootSelector, methods, children);
-      if (!this.treeMap.has(pageObjectName)) {
-        this.treeMap.set(pageObjectName, rootNode);
-      }
-    } else {
-      // For non-root Page Object
-      const childNode = new PageObjectTreeNode(pageObjectName, rootSelector, methods, children);
-      for (let [key, value] of this.treeMap) {
-        if (value.length !== 0) {
-          const existingChildren = value.children;
-          for (let i = 0; i < existingChildren.length; i++) {
-            if (existingChildren[i].name === pageObjectName) {
-              existingChildren[i] = childNode;
-              value.children = existingChildren;
-              this.treeMap.set(key, value);
-              // Get out from the inside for loop
-              i = existingChildren.length;
+              method.returnType.methods[childMethodName].Java_Code = method.Java_Code + '.' + childMethodName + arglist + ';';
+              method.returnType.methods[childMethodName].JS_Code = method.JS_Code + '.' + childMethodName + arglist + ';';
             }
           }
         }
       }
+      this.treeMap.set(poName, po);
     }
   }
 
-  parseImpls (pageObjectName, rootNode) {
-    const rootMarkerNode = findNodeAtLocation(rootNode, ['root']);
-    if (rootMarkerNode !== undefined && rootMarkerNode.value === true) {
-      const rootSelectorNode = findNodeAtLocation(rootNode, ['selector']);
-      const profileNode = findNodeAtLocation(rootNode, ['profile']);
-      if (rootSelectorNode && profileNode.children && this.treeMap.has(pageObjectName)) {
-        const { methods, children, rootSelector } = this.treeMap.get(pageObjectName);
-        const platformNode = profileNode.children[0];
-        const platform = platformNode.children[0].children[1].children[0].value.split('_')[0].toLowerCase();
-        const selector = rootSelectorNode.children[0].children[1].value;
-        if (platform === 'ios') {
-          rootSelector.ios = selector;
-        } else if (platform === 'android') {
-          rootSelector.android = selector;
+  // Build the tree map from Page Object json file
+  buildTreeMapFromFile (file, filePath) {
+    if (file.includes(UTAM_EXT)) {
+      let [pageObjectName] = basename(filePath).split('.');
+      if (pageObjectName.endsWith('Android')) {
```

Comment:
make const variable for android and ios, and use it at here, and also to replace -7 at L75 and L81